.taskcluster.yml

```diff
@@ -0,0 +1,70 @@
+version: 1
+policy:
+  pullRequests: public
+tasks:
+  $if: taskcluster_root_url == "https://taskcluster.internal.wpgapi.uk"
+  then:
+    $let:
+      imageName: gcr.io/wpg-internal/taskcluster-operator
+    in:
+      - $if: (tasks_for == "github-push") && (event.ref == "refs/heads/main")
+        then:
+          taskId:
+            $eval: as_slugid("push_task")
+          created: { $fromNow: '' }
+          deadline: { $fromNow: '3 hours' }
+          expires: { $fromNow: '10 days' }
+          provisionerId: shared
+          workerType: multistage-docker
+          scopes:
+            - secrets:get:shared/github
+            - secrets:get:shared/cloudbuild-service-account
+          routes:
+            - notify.slack-channel.C9Z43K24F.on-any
+          payload:
+            steps:
+              $flatten:
+                - image: gcr.io/wpg-internal/ci-bootstrap
+                - image: bitnami/git
+                  command: ["/bin/bash", "-c"]
+                  args:
+                    - git clone ${event.repository.html_url} . && git checkout ${event.after}
+                - image: docker
+                  args: [ "build", ".", "-t", "${imageName}:${event.after}" ]
+                - image: docker
+                  args: [ "push", "${imageName}:${event.after}" ]
+                - image: gcr.io/wpg-internal/update-infra-image
+                  args: ["deployments/packages/_taskcluster-operator/package.json=config.imageTags.taskcluster-operator=${event.after}"]
+          metadata:
+            name: Build taskcluster-operator
+            description: Build & push taskcluster-operator
+            owner: '${event.sender.login}@users.noreply.github.com'
+            source: '${event.repository.url}'
+      - $if: (tasks_for == "github-pull-request") && event["action"] in ["opened", "reopened", "synchronize"]
+        then:
+          taskId:
+            $eval: as_slugid("pr_task")
+          created: { $fromNow: '' }
+          deadline: { $fromNow: '3 hours' }
+          expires: { $fromNow: '3 days' }
+          provisionerId: shared
+          workerType: multistage-docker
+          scopes:
+            - secrets:get:shared/github
+          payload:
+            steps:
+              - image: gcr.io/wpg-internal/ci-bootstrap
+              - image: bitnami/git
+                command: ["/bin/bash", "-c"]
+                args:
+                  - |
+                    git clone ${event.repository.html_url}
+                    && git checkout ${event.pull_request.head.sha}
+                    && git rebase ${event.pull_request.base.sha}
+              - image: docker
+                args: [ "build", "." ]
```

Comment:
no push with the PR number, or is this something we're phasing out in preference of other future tooling?

Reply:
Nobody has ever used those images (unless you have)... I can't see much of an argument for keeping them...

Reply:
I technically have used them, but they immediately fall over on the fact infra/k8s can't detect changes in those images c cos they use the same image name. Technically it'd be handy for people who don't want to/cannot run my build&deploy script (or perform it manually like you do) to open a PR and use the tag to manually put in infra and do a scratch-2 deployment (for example). Not a hard requirement, just want people to have a nice workflow for this, ala forklift v2 or whatever, yadda yadda.